core/src/main/java/org/hibernate/ogm/dialect/eventstate/impl/EventContextManagerInitiator.java

```diff
@@ -30,7 +31,16 @@ private EventContextManagerInitiator() {
 	}
 
 	@Override
-	public EventContextManager initiateService(Map configurationValues, ServiceRegistryImplementor registry) {
-		return new EventContextManager();
+	public EventContextManager initiateService(SessionFactoryImplementor sessionFactory, SessionFactoryOptions sessionFactoryOptions,
+			ServiceRegistryImplementor registry) {
+		EventContextManager eventContextManager = new EventContextManager( registry );
+
+		GridDialect gridDialect = registry.getService( GridDialect.class );
+		// we don't use GridDialects.hasFacet here as we need to set it even for a {@link ForwardingGridDialect}
+		if ( gridDialect instanceof EventContextManagerAwareGridDialect ) {
```

Comment:
Why is this though? Seems `ForwardingGridDialect` is only passing it on if the wrapped dialect actually is a `EventContextManagerAwareGridDialect`?